### DISCUSSION TITLE
customFields not retrieved in Remix Storefront

### DISCUSSION THREAD

**Discussion by huwgit** [vendurehq/vendure#1924] [Q&A]
I have added the following custom fields

       Product: [
            { name: 'weight', type: 'string' },
    
            { name: 'size', type: 'localeString' },
          ]

It works fine on the admin site. On the storefront in remix I cannot access these fields even though they are they are there - I can retrieve them using graphql:

{
  product(slug: "laptop") {
    name
    id
    customFields{
      weight
    }
  }
}

Any idead where my remix storefront not picking these up?

**Comment by michaelbromley** [comment]
Hi,
A GraphQL client application like the Remix storefront will not automatically fetch any fields not explicitly specified by the queries it is using. So you'll need to add the custom fields to your field selection in the [product query](https://github.com/vendure-ecommerce/storefront-remix-starter/blob/master/app/providers/products/products.ts#L20).

**Reply by huwgit** [reply]
I have added customFields in /providers/product/product.ts but still not working:

export const detailedProductFragment = gql`
  fragment DetailedProduct on Product {
    id
    name
    description
    customFields {
      weight
      size
    }

**Reply by huwgit** [reply]
am I referencing the custom field correctly? {product.customFields.weight}

**Reply by huwgit** [reply]
on querying the product entity i get the message:

Error: Objects are not valid as a React child (found: object with keys {id, name, description, collections, facetValues, featuredAsset, assets, variants}) 

no mention of customFields...

**Reply by michaelbromley** [reply]
Oh I think you'll need to run codegen first, `npm run generate`

**Reply by huwgit** [reply]
when I run it I get

    ✖ GraphQL Document Validation failed with 1 errors;
      Error 0: Field "customFields" must not have a selection since type "JSON" has no subfiel…
      at /Users/buthenahdavies/storefront-remix-starter/app/providers/products/products.ts:6:18
  ❯ Generate to app/generated/schema.graphql

**Reply by michaelbromley** [reply]
Try changing [this line](https://github.com/vendure-ecommerce/storefront-remix-starter/blob/0bed3be60e3ce1eea12c51fee5eeba3c56ae6621/codegen.yml#L3) to point to your local Vendure server

```yaml
overwrite: true
schema:
  - "http://localhost:3000/shop-api"
```

**Reply by huwgit** [reply]
Yes that did. Thanks a lot.

**Reply by michaelbromley** [reply]
Good to know! We're going to add a guide to the docs of the Remix storefront to make this easier for others who want to do this.

**Reply by kyunal** [reply]
I went ahead and published the [guide](https://github.com/vendure-ecommerce/storefront-remix-starter/wiki/Querying-custom-fields) which covers the necessary steps 🎉

**Comment by huwgit** [comment]
That looks great, thanks once again.

Huw
